src/extension.ts

```diff
@@ -99,4 +102,24 @@ class ExtensionRuntime {
   public async dispose(): Promise<void> {
     await this.client?.stop();
   }
+
+  public async startClientAndWaitForVersion() {
+    this.client = this.client ?? await DafnyLanguageClient.create(this.context, this.statusOutput);
+    this.client.start();
+    await this.client.onReady();
+    this.languageServerVersion = await this.getLanguageServerVersionAfterStartup();
+  }
+
+  public async restart(): Promise<void> {
+    this.statusOutput.appendLine('Terminating Dafny...');
+    await this.dispose();
+    // The first subscription is the statusOutput and should not be disposed.
```

Comment:
Could we delete that one and recreate it as well?

Reply:
I did not find out how to do it properly. Activate() is called only once.